catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsLoader.java

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.forms;
+
+import ddf.catalog.data.Metacard;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.apache.commons.io.IOUtils;
+import org.boon.Boon;
+import org.codice.ddf.catalog.ui.forms.data.QueryTemplateMetacardImpl;
+import org.codice.ddf.catalog.ui.forms.data.ResultTemplateMetacardImpl;
+import org.codice.ddf.configuration.AbsolutePathResolver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Loads initial system template configuration from the file system so template defaults in
+ * distributions can vary independently.
+ */
+public class SearchFormsLoader implements Supplier<List<Metacard>> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SearchFormsLoader.class);
+
+  private static final File DEFAULT_FORMS_DIRECTORY =
+      new File(new AbsolutePathResolver("etc/forms").getPath());
+
+  private static final String FORMS_FILE_NAME = "forms.json";
+
+  private static final String RESULTS_FILE_NAME = "results.json";
+
+  private final File configDirectory;
+
+  public static Supplier<List<Metacard>> config() {
+    return new SearchFormsLoader();
+  }
+
+  public SearchFormsLoader() {
+    this(DEFAULT_FORMS_DIRECTORY);
+  }
+
+  public SearchFormsLoader(File configDirectory) {
+    this.configDirectory = configDirectory;
+  }
+
+  @Override
+  public List<Metacard> get() {
+    if (!configDirectory.exists()) {
+      LOGGER.debug("Could not locate forms directory");
+      return Collections.emptyList();
+    }
+
+    // What's our policy here? Do we want the system to blowup / make noise if this isn't possible?
```

Comment:
What about security manager issues? I don't think those should occur at runtime when the product is deployed in the wild. That's why I'm on the fence about including this check to begin with, because it can potentially mask a permissions issue.